be/src/io/tools/file_cache_microbench.cpp

```diff
@@ -0,0 +1,1382 @@
+                    }
+                    if (job.completion_tracker) {
+                        job.completion_tracker->mark_completed(key);
+                    }
+                    completed_writes++;
+                } catch (const std::exception& e) {
+                    LOG(ERROR) << "Write task failed for segment " << key << ": " << e.what();
+                }
+            }));
+        }
+
+        // 等待所有写入任务完成
+        for (auto& future : write_futures) {
+            future.get();
+        }
+        write_stopwatch.stop(); // 停止计时
+
+        // 将写入时间从纳秒转换为秒并格式化为字符串
+        double total_write_time_seconds = write_stopwatch.elapsed_time() / 1e9; // 纳秒转秒
+        job.stats.total_write_time =
+                std::to_string(total_write_time_seconds) + " seconds"; // 保存为字符串
+        job.stats.peak_write_iops = write_stats->get_peak_iops();
+        LOG(INFO) << "Total write time: " << job.stats.total_write_time << " seconds";
+
+        // 记录写入的文件信息
+        for (const auto& key : keys) {
+            size_t data_size = config.size_bytes_perfile;
+            record_file_info(key, data_size, job.job_id);
+        }
+    }
+
+    void execute_read_tasks(const std::vector<std::string>& keys, Job& job, JobConfig& config) {
+        int64_t exist_job_perfile_size = s3_file_records.get_exist_job_perfile_size_by_prefix(
+                HIDDEN_PREFIX + config.file_prefix);
+        std::vector<std::future<void>> read_futures;
+        doris::io::IOContext io_ctx;
+        doris::io::FileCacheStatistics total_stats;
+        io_ctx.file_cache_stats = &total_stats;
+        if (config.cache_type == "DISPOSABLE") {
+            io_ctx.is_disposable = true;
+        } else if (config.cache_type == "TTL") {
+            io_ctx.expiration_time = config.expiration;
+        } else if (config.cache_type == "INDEX") {
+            io_ctx.is_index_data = true;
+        } else { // default NORMAL
+            // do nothing
+        }
+        ThreadPool read_pool(config.num_threads);
+        std::vector<std::shared_ptr<IopsRateLimiter>> read_limiters;
+        read_limiters.reserve(config.num_files);
+        auto read_stats = std::make_shared<IopsStats>();
+        for (int i = 0; i < config.num_files; ++i) {
+            read_limiters.push_back(std::make_shared<IopsRateLimiter>(config.read_iops));
+        }
+        std::atomic<int> completed_reads(0);
+        doris::MonotonicStopWatch read_stopwatch; // 添加读取任务计时器
+
+        // 创建 S3 客户端配置
+        doris::S3ClientConf s3_conf = create_s3_client_conf(config);
+
+        read_stopwatch.start();
+        for (int i = 0; i < keys.size(); ++i) {
+            const auto& key = keys[i];
+            read_futures.push_back(read_pool.enqueue([&, key, i]() {
+                try {
+                    if (job.completion_tracker) {
+                        job.completion_tracker->wait_for_completion(key); // 等待文件完成
+                    }
+                    doris::io::FileReaderOptions reader_opts;
```

Comment:
init file_size if possible, because we known the file size.

Reply:
fixed